addons/account/models/partner.py

```diff
@@ -213,12 +213,19 @@ def _get_vat_valid(self, delivery, company=None):
         """ Hook for determining VAT validity with more complex VAT requirements """
         return bool(delivery.vat)
 
+    def _check_current_company_have_auto_apply(self):
+        return self.search_count([('company_id', '=', self.env.company.id), ('auto_apply', '=', True)], limit=1) > 0
+
     def _prepare_fpos_base_domain(self, vat_required):
-        return [
-            *self._check_company_domain(self.env.company),
+        domain = [
             ('auto_apply', '=', True),
             ('vat_required', '=', vat_required),
         ]
+        if self._check_current_company_have_auto_apply():
+            domain.append(('company_id', '=', self.env.company.id))
+        else:
+            domain.extend(self._check_company_domain(self.env.company))
+        return domain
```

Comment:
@Levizar Nice catch and Thanks for the suggestions!
Can you please explain the case when it goes to fallback (with no state and no zip)? if that is the case shouldn't it be the first domain search it would be? Still thinking about the cases where we can still optimize it :thinking: